task.go

```diff
@@ -283,14 +283,17 @@ func (t *task) Delete(ctx context.Context, opts ...ProcessDeleteOpts) (*ExitStat
 	if t.io != nil {
 		t.io.Cancel()
 		t.io.Wait()
-		t.io.Close()
 	}
 	r, err := t.client.TaskService().Delete(ctx, &tasks.DeleteTaskRequest{
 		ContainerID: t.id,
 	})
 	if err != nil {
 		return nil, errdefs.FromGRPC(err)
 	}
+	// Only close the IO after a successful Delete
```

Comment:
nit: `s/close/cleanup` ?

If it was just closing the IO it wouldn't need to be done so late after a `Cancel`/`Wait` combo.

Reply:
Done.